Cpp_test/lib/ipc-shm.cpp

```diff
@@ -0,0 +1,113 @@
+#include "ipc-shm.h"
+
+IPCShm::~IPCShm()
+{
+
+	if (pthread_condattr_destroy(&(this->cond_attr)) != 0)
```

Comment:
what is  happenning if the condvar is not set up? Same question for `pthread_mutexattr_destroy`.

Reply:
It will return the error number: EINVAL, which is cond is invalid.